File: 0108-convert-sorted-array-to-binary-search-tree/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) {
        return helper(nums, 0, nums.length-1);
        
    }

    public  TreeNode helper(int []nums, int s, int e){

        if(s>e) return null;
        if(s==e) return new TreeNode(nums[s]);

        int mid = (s+e)/2;

        TreeNode newNode = new  TreeNode(nums[mid]);

        newNode.left = helper(nums,s,mid-1);
        newNode.right = helper(nums, mid+1, e);

        return newNode;
    }
}

File: 0108-convert-sorted-array-to-binary-search-tree/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    TreeNode* helper(vector<int>& nums, int start, int end){
        if(start > end) return NULL;
        if(start == end){
            TreeNode* node = new TreeNode(nums[start]);
            return node;
        }

        int mid = start + (end-start)/2;
        TreeNode* node = new TreeNode(nums[mid]);

        node->left = helper(nums, start, mid-1);
        node->right = helper(nums, mid+1, end);

        return node;
    }
public:
    TreeNode* sortedArrayToBST(vector<int>& nums) {
        return helper(nums, 0, nums.size()-1);
    }
};
